libbeat/docs/processors-using.asciidoc

```diff
@@ -624,6 +627,89 @@ _Openstack Nova_
 }
 -------------------------------------------------------------------------------
 
+[[add-fields]]
+=== Add fields
+
+The `add_fields` processor adds additional fields to the event.  Fields can be
+scalar values, arrays, dictionaries, or any nested combination of these.  By
+default the fields that you specify will be grouped under the `fields`
+sub-dictionary in the event. To group the fields under a different
+sub-dictionary, use the `target` setting. To store the fields as
+top-level fields, set `target: ''`. 
+
+`target`:: (Optional) Sub-dictionary to put all fields into. Defaults to `fields`.
+`fields`:: Fields to be added.
+
+
+For example, this configuration:
+
+[source,yaml]
+------------------------------------------------------------------------------
+processors:
+- add_fields:
+    target: project
+    fields:
+      name: myproject
+      id: '574734885120952459'
+------------------------------------------------------------------------------
+
+Adds these fields to any event:
+
+[source,json]
+-------------------------------------------------------------------------------
+{
+  "project": {
+    "name": "myproject",
+    "id": "574734885120952459"
+  }
+}
+-------------------------------------------------------------------------------
+
+
+[[add-labels]]
+=== Add labels
+
+The `add_labels` processors adds a set of key-value pairs to an event.
+The processor will flatten nested configuration objects like arrays or
+dictionaries into a fully qualified name by merging nested names with a `.`.
+Array entries create numeric names starting with 0.  Labels are always stored
+under the Elastic Common Schema compliant `labels` sub-dictionary.
+
+`labels`:: dictionaries of labels to be added.
+
+For example, this configuration:
+
+[source,yaml]
+------------------------------------------------------------------------------
+processors:
+- add_labels:
+    labels:
+      number: 1
+      with.dots: test
+      nested:
+        with.dots: nested
```

Comment:
Aren't we moving away from dots in field names, to avoid conceptual conflicts with dots representing nested objects?